lib/std/child_process.zig

```diff
@@ -1353,23 +1353,16 @@ test "build and call child_process" {
     var it = try std.process.argsWithAllocator(allocator);
     defer it.deinit(); // no-op unless WASI or Windows
     const testargs = try testing.getTestArgs(&it);
-
     var tmp = testing.tmpDir(.{ .no_follow = true }); // ie zig-cache/tmp/8DLgoSEqz593PAEE
     defer tmp.cleanup();
-    const tmpdirpath = try tmp.getFullPath(allocator);
-    defer allocator.free(tmpdirpath);
     const child_name = "child"; // no need for suffixes (.exe, .wasm) due to '-femit-bin'
-    const suffix_zig = ".zig";
-    const child_path = try fs.path.join(allocator, &[_][]const u8{ tmpdirpath, child_name });
-    defer allocator.free(child_path);
-    const child_zig = try mem.concat(allocator, u8, &[_][]const u8{ child_path, suffix_zig });
-    defer allocator.free(child_zig);
-
-    try tmp.dir.writeFile("child.zig", childstr);
-    try testing.buildExe(testargs.zigexec, child_zig, child_path);
+    const zigfile_path = try tmp.writeZigFile(allocator, childstr, child_name);
+    defer allocator.free(zigfile_path);
 
+    const binary = zigfile_path[0 .. zigfile_path.len - 4]; // '.zig' is 4 characters
+    try testing.buildExe(testargs.zigexec, zigfile_path, binary);
```

Comment:
The path to binary files can be calculated as:
```zig
const filename = std.fs.path.basename(zigfile_path);
const binary = :blk {
    const index = std.mem.lastIndexOfScalar(u8, filename, '.') orelse break :blk filename;
    if (index == 0) break :blk filename;
    break :blk filename[0..index];
}
```

This is a code snippet I created after modifying std.fs.path.extension() but just reversed and I succesfully use it in a project. It is as reliable as stdlib itself. This code can also be put inside buildExe in order to automatically deduce the binary name.

Reply:
yes, however thats 6 LOC instead of 1 and we know that `.zig` is always encoded as ascii (cause we generated) and thus 4 bytes.

I can see the usage of a function instead of `const binary = zigfile_path[0 .. zigfile_path.len - 4]; // '.zig' is 4 characters` ie for static,dynamic etc libraries or other stuff.

Your suggestion belongs into a separate PR for `path.zig` as complement to 
the function `extension`. Maybe this is your next PR? ;-)

My reasoning for simplifying this is to have readable+reasonable short tests for testing the test runner (output).

Reply:
This is true, I was just suggesting a general purpose solution.